hack/versions/update_versions.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+import os
+import subprocess
+
+from types import SimpleNamespace
+
+from ruamel.yaml import YAML
+
+yaml = YAML()
+
+from rich import print
+
+# ====
+
+CHARTS = ["charts/calico/values.yaml", "charts/tigera-operator/values.yaml"]
+
+
+def get_git_root():
+    return (
+        subprocess.check_output(["git", "rev-parse", "--show-toplevel"])
+        .strip()
+        .decode()
+    )
+
+
+def get_git_describe():
+    result = (
+        subprocess.check_output(["git", "describe", "--tags", "--long"])
+        .strip()
+        .decode()
+    )
+    describe = result.split("-")
+    version, commits, abbr = describe
+
+    d = SimpleNamespace(
+        version=version, commits=int(commits), abbr=abbr, commit=abbr.strip("g")
+    )
+    return d
+
+
+def parse_version(version):
+    return [int(ver) for ver in version.strip("v").split(".")]
+
+
+def unparse_version(version):
+    return f"v{version[0]}.{version[1]}.{version[2]}"
+
+
+def bump_version_string(version_string):
+    version = parse_version(version_string)
+    version[2] += 1
```

Comment:
I guess this means that this is only used for patch releases? Should we have an option to pass in that will allow us to use this for minor releases as well?

Reply:
It should be easy enough to detect when we're on a new minor release based on the branch name (e.g. `release-v3.29` not matching the major/minor from the latest tag name, which would currently be v3.19.0).

I'll definitely add this as a feature, but it's outside of the scope of implementation right now.